### PR TITLE
Llama3.2 knowledge distillation config

### PR DESCRIPTION
**Pull request by lindawangg**
#### Context
What is the purpose of this PR? Is it to
- [x] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

#### Changelog
What are the changes made in this PR?
* Knowledge distillation config to distill LLAMA3 models, specifically Llama 3.1 8B into Llama 3.2 1B

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [x] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [x] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [x] run unit tests via `pytest tests`
- [x] run recipe tests via `pytest tests -m integration_test`
- [x] manually run any new or modified recipes with sufficient proof of correctness
- [x] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)
```
tune run knowledge_distillation_single_device --config llama3_2/knowledge_distillation_single_device
```
image
imageimage

Eval results show that using KD yields slightly better results than just standard fine-tuning.
image

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [x] I have added an example to docs or docstrings

### CODE REVIEW

File: recipes/configs/llama3_2/knowledge_distillation_single_device.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# Config for single device knowledge distillation in kd_single_device.py
```
Comment (ebsmothers) [inline recipes/configs/llama3_2/knowledge_distillation_single_device.yaml:1]: nit
```suggestion
# Config for single device knowledge distillation in knowledge_distillation_single_device.py
```

**Review by ebsmothers — COMMENTED** [review]
This looks great! Can you also add numbers to the PR summary on peak memory and tokens/sec you see when running with this config? Modulo that I think this is good to go though

**Review by ebsmothers — APPROVED** [review]
Looks great, thanks for adding this!
